### ISSUE TITLE
Add an option to check multiline template literals with `no-trailing-spaces`

### ISSUE THREAD

**Issue by ark120202** [eslint/eslint#12626]
**What rule do you want to change?**
[no-trailing-spaces](https://eslint.org/docs/rules/no-trailing-spaces)

**Does this change cause the rule to produce more or fewer warnings?**
More

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**

```js
const value = `
	foo··
	bar→	→	
`;
```

**What does the rule currently do for this code?**
No warnings

**What will the rule do after it's changed?**
Produce 2 warnings.

Even though removing these whitespaces may change code behavior, it's quite common to have editors configured to trim trailing whitespaces, making it possible to add code with trailing whitespaces in one commit and get extra diffs when someone else edits it. Also, I think it's clearer to use escapes if it's actually intended to make it easier to notice during review.

I'm not sure what would be an expected behavior for auto-fix (remove or escape), maybe it would make sense to use new suggestions feature there?

**Are you willing to submit a pull request to implement this change?**
Yes

**Comment by kaicataldo** [comment]
Sorry that we missed this issue. I'm going to reopen it for discussion. That being said, I'm 👎 for this proposal.

> Even though removing these whitespaces may change code behavior, it's quite common to have editors configured to trim trailing whitespaces, making it possible to add code with trailing whitespaces in one commit and get extra diffs when someone else edits it. Also, I think it's clearer to use escapes if it's actually intended to make it easier to notice during review.

This is true, but I also don't think it should be ESLint's responsibility to work around this editor behavior. I do think this could be useful - just that it doesn't belong in core.

**Comment by mdjermanovic** [comment]
Reopening, based on the last comment.
